axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+
+    public void configure(Configurer configurer) {
+        configurer.defaultConfiguration()
+                  .configureSpanFactory(c -> OpenTelemetrySpanFactory.builder().build());
+    }
+}
+```
+
+Note that when not using Spring boot, tracing each message handler invocation is not supported due to a limitation.
+
+## OpenTracing <a id="opentracing"></a>
+
+OpenTracing is deprecated. If necessary, you can
+use [the OpenTracing extension to Axon Framework](../../extensions/tracing.md).
+The functionality of the extension is rather limited compared to the OpenTelemetry integration, so using that if
+possible is preferred.
+
+## Logging <a id="logging"></a>
+
+Sometimes you don't have an APM system available, for instance during local development. It might still be useful to see
+the traces that would be started and finished to obtain insights. For this purpose, the framework provides
+a `LoggingSpanFactory`.
+
+You can configure the `LoggingSpanFactory` in the following ways:
+{% tabs %}
+{% tab title="Spring Boot Bean" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Bean
+    public SpanFactory spanFactory() {
+        return LoggingSpanFactory.INSTANCE;
+    }
+}
+```
+
+{% endtab %}
+{% tab title="Manual" %}
+
+```java
+public class AxonConfigurer {
+
+    public void configure(Configuration configuration) {
+        configuration.configureSpanFactory(c -> LoggingSpanFactory.INSTANCE);
+    }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Traced components
+
+Axon Framework provides a large range of components that are traced by the configured `SpanFactory`.
+Each component and their added spans are available for reference in this section, with additional information about how
+they should be interpreted.
+
+It's important to note that the availability of these spans is highly dependent on the application configuration. 
+For instance, some components are only used when using Axon Server, or you might have created your own `CommandBus` 
+implementation which does not call the `SpanFactory` API. 
+
+### Commands
+
+The `CommandBus` is instrumented to create spans for both dispatching and handling commands. 
+
+When using the `AxonServerCommandBus`, there will be two handling and dispatch traces 
+since it uses a second `CommandBus` to invoke the command locally after receiving it from Axon Server. In addition,
+you can see the GRPC-call to Axon Server and the time it took to handle the call.
+
+The traces will, in sequence, look like this:
+
+| Trace name                                      | Description                                                                                                                                                                    |
+|-------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```

Comment:
Just a though, but how about we use those code blocks to provide the different scenarios?
So, one with AxonServer, one without, one with the `DistributedCommandBus` (in the future), etc.

Reply:
I do like the idea and will apply it. I am, however, very afraid this will blow up to a lof of tabs